### DISCUSSION TITLE
Where are the implementations for shell-related Actions?

### DISCUSSION THREAD

**Discussion by bfahrenfort** [microsoft/terminal#12969] [Q&A]
Hi,

I'm looking at issues #12860 and #12859, and want to know where someone would implement their behaviors. Sorry if it's a dumb question, thanks in advance!

**Comment by zadjii-msft** [comment]
Heyo! Sorry for just seeing this now, still getting used to Discussions.

Based on the comments in #12859, I'm guessing you sorta figured this one out. Most of our actions are handled in `AppActionHandlers.cpp`, with a few of the older ones calling into methods that are defined in `TerminalPage.cpp`.

There are some prior examples of calling `ShellExecute` in `TerminalPage.cpp`, and we generally copy to the clipboard in `TerminalPage::_CopyToClipboardHandler`. 

Lemme know if you get blocked anywhere else, or anything else needs clarification!
